src/plugins/autocompletion_providers/Vagrant.ts

```diff
@@ -0,0 +1,40 @@
+import {PluginManager} from "../../PluginManager";
+import {linedOutputOf} from "../../PTY";
+import {styles, Suggestion, contextIndependent} from "./Common";
+
+const commands = contextIndependent(async() => {
+    let lines: string[] = [];
+
+    try {
+        lines = await linedOutputOf("vagrant", ["list-commands"], process.env.HOME);
+    } catch (e) {
+        // skip if command is not exist
```

Comment:
There is a function `executablesInPaths` to check if you have this executable. It requires `EnvironmentPath`, though, which makes it not `contextIndependent` anymore. But it's OK.